File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

/**
 * 品牌接口
 */
public interface BrandService {

    /**
     * 查询所有品牌
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * 分页查询
     * @param currentPage 当前页
     * @param pageSize 每页显示条数
     * @return
     */
    public PageResult findPage(int currentPage,int pageSize);


    /**
     * 增加数据
     * @param brand
     */
    public void add(TbBrand brand);

    /**
     * 通过id查询
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 修改品牌信息
     * @param brand
     */
    public void update(TbBrand brand);

    /**
     * 删除品牌信息
     * @param ids
     */
    public void delete(Long[] ids);


    /**
     * 分页模糊查询
     * @param currentPage 当前页
     * @param pageSize 每页显示条数
     * @return
     */
    public PageResult findPage(TbBrand brand,int currentPage,int pageSize);

    /**
     * 查询品牌名称和id
     * @return
     */
    public List<Map> selectOptionList();
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
//创建服务层
app.service("brandService",function ($http) {
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }

    this.findPage=function (currentPage, pageSize) {
        return $http.get("../brand/findPage.do?currentPage="+currentPage+"&pageSize="+pageSize);
    }

    this.save=function (entity,methodName) {
        return $http.post("../brand/"+methodName+".do",entity);
    }

    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    }

    this.dele=function (selectId) {
        return $http.get('../brand/delete.do?ids='+selectId);
    }
    this.search=function (currentPage,pageSize,searchEntity) {
        return $http.post("../brand/search.do?currentPage="+currentPage+"&pageSize="+pageSize,searchEntity)
    }

    this.findBrandList=function () {
        return $http.get("../brand/selectOptionList.do");
    }
});